### PR TITLE
client: Events for player Skull and Prayer Icon Changes

### PR DESCRIPTION
**Pull request by stone-wall**

### CODE REVIEW

**Comment by xKylee** [comment]
LGTM

File: runelite-api/src/main/java/net/runelite/api/events/player/headicon/OverheadPrayerChanged.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package net.runelite.api.events.player.headicon;
+
+import lombok.Getter;
+import net.runelite.api.HeadIcon;
+import net.runelite.api.Player;
+
```
Comment (ThatGamerBlue) [inline runelite-api/src/main/java/net/runelite/api/events/player/headicon/OverheadPrayerChanged.java]: remove constructor, add `@Data`, dont extend PlayerHeadIconChanged, just add the actor in

File: runelite-api/src/main/java/net/runelite/api/events/player/headicon/PlayerHeadIconChanged.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package net.runelite.api.events.player.headicon;
+
+import lombok.Getter;
+import net.runelite.api.Player;
+import net.runelite.api.events.Event;
+
```
Comment (ThatGamerBlue) [inline runelite-api/src/main/java/net/runelite/api/events/player/headicon/PlayerHeadIconChanged.java]: delete this class

File: runelite-api/src/main/java/net/runelite/api/events/player/headicon/PlayerSkullChanged.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package net.runelite.api.events.player.headicon;
+
+import lombok.Getter;
+import net.runelite.api.Player;
+import net.runelite.api.SkullIcon;
+
```
Comment (ThatGamerBlue) [inline runelite-api/src/main/java/net/runelite/api/events/player/headicon/PlayerSkullChanged.java]: remove constructor, add `@Data`, dont extend PlayerHeadIconChanged, just add the actor in

File: runelite-api/src/main/java/net/runelite/api/events/player/headicon/OverheadPrayerChanged.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class OverheadPrayerChanged extends PlayerHeadIconChanged
+{
+	@Getter
+	private final HeadIcon oldHeadIcon;
+
+	@Getter
+	private final HeadIcon newHeadIcon;
+
+	public OverheadPrayerChanged(Player player, HeadIcon changedFrom, HeadIcon changedTo)
```
Comment (ThatGamerBlue) [inline runelite-api/src/main/java/net/runelite/api/events/player/headicon/OverheadPrayerChanged.java]: change Player -> Actor

Comment (stone-wall) [reply runelite-api/src/main/java/net/runelite/api/events/player/headicon/OverheadPrayerChanged.java]: the event is only produced from the RSPlayer class atm cuz I didnt wanna half to deal with instanceof checks when handling it but i now realize that is useful for npcs too.

should i move it to RSActor or add a separate event for NPcs?

**Pull request merged**